gadget/ondisk.go

```diff
@@ -363,6 +363,23 @@ func BuildPartitionList(dl *OnDiskVolume, pv *LaidOutVolume) (sfdiskInput *bytes
 	return buf, toBeCreated
 }
 
+// UpdatePartitionList re-reads the partitioning data from the device and
+// updates the partition list in the specified volume.
+func UpdatePartitionList(dl *OnDiskVolume) error {
```

Comment:
Is there an open PR where there's an example use of this helper? I'm not sure I see why we'd call UpdatePartitionList() which calls OnDiskVolumeFromDevice() internally and then replace some fields. Why not take the whole new disk volume that OnDiskVolumeFromDevice returned?

Reply:
The helper is used in this PR after we remove partitions created during the install process after we alter the disk's partition table. We updated the modified fields to be explicit about what is the relevant change, but we could have used the whole new disk, all the data except the new partition table and structures should be identical.

Reply:
keeping it as a function seems the prudent thing at this point. I'm not sure it can become a methond once we merge the structures. Maybe it can be a method but on something else.